integration/nri_test.go

```diff
@@ -0,0 +1,1184 @@
+		WithCommand("/bin/sh", "-c", "sleep 3600"),
+	)
+
+	time.Sleep(1 * time.Second)
+
+	chk, err := waitForFileAndRead(filepath.Join(out, "result"), time.Second)
+	require.NoError(t, err, "read result")
+
+	expected := "Mems_allowed_list:\t" + availableMemset[1] + "\n"
+	require.Equal(t, expected, string(chk), "check result")
+}
+
+// An NRI test along with its state.
+type nriTest struct {
+	t         *testing.T
+	name      string
+	prefix    string
+	runtime   cri.RuntimeService
+	plugins   []*mockPlugin
+	namespace string
+	sbCfg     map[string]*runtime.PodSandboxConfig
+	pods      []string
+	ctrs      []string
+}
+
+// setup prepares the test environment, connects all NRI plugins.
+func (tc *nriTest) setup() {
+	if tc.name == "" {
+		tc.name = tc.t.Name()
+	}
+	if tc.prefix == "" {
+		tc.prefix = strings.ToLower(tc.name)
+	}
+	if tc.namespace == "" {
+		tc.namespace = tc.prefix + "-" + fmt.Sprintf("%d", os.Getpid())
+	}
+
+	tc.sbCfg = make(map[string]*runtime.PodSandboxConfig)
+	tc.runtime = runtimeService
+
+	for idx, p := range tc.plugins {
+		p.namespace = tc.namespace
+
+		if p.logf == nil {
+			p.logf = tc.t.Logf
+		}
+		if p.idx == "" {
+			p.idx = fmt.Sprintf("%02d", idx)
+		}
+
+		err := p.Start()
+		require.NoError(tc.t, err, "start plugin")
+
+		err = p.Wait(PluginSynchronized, time.After(pluginSyncTimeout))
+		require.NoError(tc.t, err, "wait for plugin setup")
+	}
+}
+
+// Connect a new NRI plugin to the runtime.
+func (tc *nriTest) connectNewPlugin(p *mockPlugin) {
+	p.namespace = tc.namespace
+
+	if p.logf == nil {
+		p.logf = tc.t.Logf
+	}
+	if p.idx == "" {
+		p.idx = fmt.Sprintf("%02d", len(tc.plugins))
+	}
+
+	tc.plugins = append(tc.plugins, p)
+
+	err := p.Start()
+	require.NoError(tc.t, err, "start plugin")
+
+	err = p.Wait(PluginSynchronized, time.After(pluginSyncTimeout))
+	require.NoError(tc.t, err, "wait for plugin setup")
+}
+
+// Create a pod in a namespace specific to the test case.
+func (tc *nriTest) runPod(name string) string {
+	id, cfg := PodSandboxConfigWithCleanup(tc.t, name, tc.namespace)
+	tc.sbCfg[id] = cfg
+	tc.pods = append(tc.pods, id)
+	return id
+}
+
+// Start a container in a (testcase-specific) pod.
+func (tc *nriTest) startContainer(podID, name string, opts ...ContainerOpts) string {
+	podCfg := tc.sbCfg[podID]
+	require.NotNil(tc.t, podCfg, "pod config for "+podID)
+
+	ctrCfg := ContainerConfig(name, "", opts...)
+	if ctrCfg.Image.Image == "" {
+		image := images.Get(images.BusyBox)
+		EnsureImageExists(tc.t, image)
+		ctrCfg.Image.Image = image
+	}
+
+	if len(ctrCfg.Command) == 0 {
+		WithCommand("/bin/sh", "-c", "sleep 3600")(ctrCfg)
+	}
+
+	ctrID, err := tc.runtime.CreateContainer(podID, ctrCfg, podCfg)
+	assert.NoError(tc.t, err, "create container")
+	assert.NoError(tc.t, tc.runtime.StartContainer(ctrID), "start container")
+
+	tc.ctrs = append(tc.ctrs, ctrID)
+
+	tc.t.Cleanup(func() {
+		assert.NoError(tc.t, tc.runtime.StopContainer(ctrID, containerStopTimeout))
```

Comment:
? not sure where this version StopContainer is that receives a const..as the second variable..

Reply:
It's `integration/remote/[New]RuntimeService[()]` which implements `integration/cri-api/pkg/apis/ContainerManager`. It is used throughout the integration tests to create/start/stop/remove/etc. test containers. I simply monkey-see-monkey-do-adopted that from the other existing tests.

Reply:
thx forgot about the integration/remote..